### ISSUE TITLE
NT 2020.006 - Intermediarios - prazo produção 05/04/2021

### ISSUE THREAD

**Issue by robmachado** [nfephp-org/sped-nfe#783]
Novos campos inclusos

Inclusão do campo de indicativo da operação com intermediador/marketplace, que será obrigatório
informar quando o indicador de presença for 2=Operação não presencial, pela Internet; 3=Operação
não presencial, Teleatendimento; 4=NFC-e em operação com entrega a domicílio; ou 9=Operação não
presencial, outros.

Inclusão de campos com as informações do intermediador da transação: CNPJ do Intermediador da
Transação, Identificador Cadastro Intermediador

Novos Modos de pagamento
16=Depósito Bancário
17=Pagamento Instantâneo (PIX)
18=Transferência bancária, Carteira Digital
19=Programa de fidelidade, Cashback, Crédito Virtual
98=Regime Especial NFF

**Comment by robmachado** [comment]
Liberado para testes quando estiver disponivel

**Comment by MarconWillian** [comment]
Amigo, como faço para testar estes update que você fez no teu fork?

**Comment by robmachado** [comment]
Prezado !! isso é uma coisa que você já deveria saber fazer se criou um fork !! 
vou te responder mas não pergunte novamente sobre esse tipo de assunto !! 
E caso tenha feito ALTERAÇÕES no seu fork, provavelmente não poderá mesclar com o origem.

git remote add upstream git@github.com:nfephp-org/sped-nfe.git
git fetch upstream
git merge upstream/master

**Comment by robmachado** [comment]
Lia a documentação da SEFAZ para detalhes !!
Para testar inclua os campos 

tagIde()  - campo adicional : indIntermed

$std = new stdClass();
$std->CNPJ = '12345678901234';
$std->idCadIntTran = 'Fulano'; 

tagIntermed($std) ;

**Comment by MarconWillian** [comment]
Sim sim, esta parte toda eu já sabia, desculpa se me coloquei errado, mas pensei que tu tinha subido o package como dev ou algo assim, desculpa se fui inconveniente.

**Comment by robmachado** [comment]
Não colega !! tudo bem sem problemas !! eu sou meio chato mesmo ... pois me fazem perguntas que doi !!
Esperto que a explicação sobre o teste tenha sido suficiente !

**Comment by MarconWillian** [comment]
Foi sim, eu normalmente não faço edições no fork principal, dai vou dar um merge la no meu, muito obrigado.

**Comment by GustavoBc89** [comment]
Bom dia, fiz a implementação da Tag indIntermed no sistema, mas ao tentar reenviar a nota da um erro, como se essa tag não estive sendo reconhecida, sabem me dizer se essa tag já foi implementada no projeto nfe-sped?

**Comment by robmachado** [comment]
Está implementado SIM, testado e sendo usado em homologação.
Agora esse tipo de informação não ajuda ninguém a te ajudar. "dá um erro" não é útil !! 
NOTA: algumas SEFAZ não atualizaram seus servidores AINDA !! funcionarios publicos o que esperar de um governo caótico em todos os níveis !

**Comment by robmachado** [comment]
Está apenas em MASTER !! não está em nenhuma release ainda

**Comment by GustavoBc89** [comment]
Entendi, mais uma vez muito obrigo pela retorno e atenção.

**Comment by robmachado** [comment]
Sem problema !

**Comment by rabraga** [comment]
@robmachado, bom dia!
A SEFAZ nesta NT está rejeitando em produção essa nova tag intermed. Vai aceitar só depois de 05/04, o mesmo serve para os novos meios de pagamento. Em homologação eu já fiz um teste e foi autorizada. 

Em outras NT que obrigava uma nova tag não acontecia a rejeição antes da data, só rejeitava se não informava após a data. 

Uma verdadeira bosta. Agora tem que implementar colocando datas no código fonte. 

Como que vai ser feito com relação aos eschemas, já que essa tag é obrigatória, como será ao assinar e validar o XML?

**Comment by rabraga** [comment]
@robmachado , vi aqui que no \schemes\PL_009_V4\leiauteNFe_v4.00.xsd não é obrigatório, então não vai ter problemas. 
xs:element name="indIntermed" minOccurs="0"

**Comment by robmachado** [comment]
Se o campo não for passado ou estiver NULO não será incluso no XML.
Mas se contiver ZERO ou string vazia vai ser incluso ! E nesse caso dá erro em produção !

**Comment by rxsantana** [comment]
A SEFAZ vai liberar o envio das novas tags a partir de 05/04/2021, mas não será obrigatório, ou seja, eles não vão validar se existe ou não as tags. 
Esta validação só vai começar a ser realizada a partir de setembro, com isso, temos a tranquilidade de atualizar os clientes aos poucos a partir do dia 05/04.

**Comment by robmachado** [comment]
Como sempre eles nos deixam mais e mais confusos !!

**Comment by robmachado** [comment]
@rxsantana Ricardo **de onde obteve essas informações** que não haverá validação desses campos até setembro !!

**Comment by rxsantana** [comment]
@robmachado , bom dia!

Obtive a informação na Nota Técnica 2020.006 - v.1.10. Página 9.

Rejeição: 434 - NF-e sem indicativo do intermediador

"Observação 2: Regra válida para Nota Fiscal Avulsa eletrônica a partir de 05/04/2021 para homologação e 01/09/2021 para produção"

**Comment by robmachado** [comment]
Rapaz !! deixei isso passar, tava meio escondido e eu não li o documento de forma atenta !!

Muito obrigado pela informação e ajuda :+1:

**Comment by robmachado** [comment]
Então tem um detalhe ai !!  pegadinha !!
**Regra válida para Nota Fiscal Avulsa eletrônica** 
Apenas se refere a NFA ou seja aquelas NFe criadas na SEFAZ !!

**Comment by robmachado** [comment]
NT 2020,006 - v1.10 pagina 9

_Se Informado indicativo de presença, tag: indPres, IGUAL a 1, 2, 3, 4 ou 9_
- Obrigatório o preenchimento do campo Indicativo do Intermediador (tag: indIntermed)
**Observação 1: Regra válida a partir de 01/02/2021 para homologação e 01/09/2021 para produção**
Observação 2: Regra válida para **Nota Fiscal Avulsa eletrônica** a partir de 05/04/2021 para homologação e 01/09/2021 para produção

**Comment by rabraga** [comment]
É uma palhaçada atrás da outra esta SEFAZ.

**Comment by robmachado** [comment]
Para os ambientes autorizadores **SVAN, SP, GO, MG** os prazos previstos para a Nota Técnica 2020.006 v1.00 e v1.10 são:

- Ambiente de Homologação (ambiente de teste das empresas): 01/03/2021
- Ambiente de Produção: 05/04/2021

Mas acredito que em função da Observação 1 - a validação só deverá ocorrer a partir de 01/09/2021, assim espero mas é **difícil saber**, tem estados que já colocaram em homologação e outros ainda não uma verdadeira confusão !!

**Comment by robmachado** [comment]
PR por exemplo dá erro caso não tenha o campo "indIntermed" na tag ide da NFe

**Comment by rabraga** [comment]
@robmachado , pra mim não ficou muito claro se é somente para NFA. Veja que as datas são diferentes para NFa em homologação. Ou seja, começa a validar em homologação para NFA quando for para produção em NFe.

**Comment by robmachado** [comment]
**Observação 1: Regra válida a partir de 01/02/2021 para homologação e 01/09/2021 para produção**
Refere-se as NFe/NFCe emitidas por nós

**Observação 2: Regra válida para Nota Fiscal Avulsa eletrônica a partir de 05/04/2021 para homologação e 01/09/2021 para produção**
Refere-se as NFe/NFCe emitidas pela SEFAZ

**Comment by robmachado** [comment]
Mas podem haver estados que operem diferentemente vai saber !!
Eu não confio nessas informações faz tempo !!

**Comment by rodrigojob** [comment]
> Para os ambientes autorizadores **SVAN, SP, GO, MG** os prazos previstos para a Nota Técnica 2020.006 v1.00 e v1.10 são:
> 
> * Ambiente de Homologação (ambiente de teste das empresas): 01/03/2021
> * Ambiente de Produção: 05/04/2021
> 
> Mas acredito que em função da Observação 1 - a validação só deverá ocorrer a partir de 01/09/2021, assim espero mas é **difícil saber**, tem estados que já colocaram em homologação e outros ainda não uma verdadeira confusão !!

Eu fiz um procedimento enorme para corrigir um monte de coisas, ao emitir a NF deu erro de schema. Claro, além de obrigatório a partir dos dia 5/4, só apartir do dia 5/4 que será aceito com estes dados!!!?
Deveria aceitar antes, mas não validar.

**Comment by rabraga** [comment]
Parece piada... Mas é só a SEFAZ sendo SEFAZ...
Hoje 05/04 rejeitou o envio da tag indIntermed, retornando rejeição 225 falha no XML.

**Comment by gersonfs** [comment]
Verdade, notem que teve uma alteração nessa NT republicada em 17/03. Agora deve entrar em produção em 01/09
Primeiro era opcional informar indIntermed em alguns casos, como quando indPres = 1. Depois foi tornado obrigatório, e agora é opcional novamente quando indPres = 1 
Tem tag nova nos pagamentos, a YA02a - xPag

**Comment by gersonfs** [comment]
> Parece piada... Mas é só a SEFAZ sendo SEFAZ...
> Hoje 05/04 rejeitou o envio da tag indIntermed, retornando rejeição 225 falha no XML.

Verdade, notem que teve uma alteração nessa NT republicada em 17/03. Agora deve entrar em produção em 01/09
Primeiro era opcional informar indIntermed em alguns casos, como quando indPres = 1. Depois foi tornado obrigatório, e agora é opcional novamente quando indPres = 1
Tem tag nova nos pagamentos, a YA02a - xPag

**Comment by rabraga** [comment]
Veja a inclusão do texto na NT liberada Republicada em 17/03/2021:
* Os campos B25c (indIntermed) e o grupo de intermediador (infIntermed) YB01 estarão disponíveis a partir 05/04/2021 em produção, porém, a validação ocorrerá somente a partir do dia 01/09/2021.

**Comment by joaoroyosilva** [comment]
Alguém sabe se os meios de pagamento já podem ser utilizados?

**Comment by robmachado** [comment]
O prazo previsto para a Nota Técnica 2020.006 v1.20 é:

- Ambiente de Homologação (ambiente de teste das empresas): até 03/05/2021
- Ambiente de Produção: 01/09/2021 


**Mas isto ainda está extremamente confuso !!!** desta vez a SEFAZ criou um problema enorme de múltiplas datas, e sem muita consistência.

Então só DEUS sabe a verdade !

**Comment by devsabrinaelias** [comment]
Olá, eu abaixei todos os arquivos e pastas para começar a fazer os ajustes da Nota Técnica 2020.006.

Estou na parte onde se eu informar 99 na tag TPag eu tenho que preencher a tag xPag, eu estou fazendo isso mas esta me retornado uma rejeição: 436 - Motivo: Rejeição: Informado 99-Outros como meio de pagamento. Mas no PDF da nota técnica página 10/12 essa regra de validação está riscada. Os arquivos estão atualizados a respeito disso?

Desde já grata.

**Comment by robmachado** [comment]
leia o meu post anterior !! por favor ! 
E sim a biblioteca já inclui essas informações (veja o xml que foi gerado por você pela classe Make())
Esse erro não é da biblioteca e sim da SEFAZ
Essa rejeição ocorre quando é emitida uma NF-e (modelo 55) ou NFC-e (modelo 65) e no Tipo de Pagamento foi informado um valor que não consta na tabela de meios de pagamento disponibilizada pela Sefaz (ou seja ainda não aceita 99 !!)

**Comment by devsabrinaelias** [comment]
Olá boa tarde, será podem me ajudar a esclarecer uma dúvida? Essa Nota Técnica 2020.006 v1.20  eu já posso colocar ela em produção ou ela só vai ficar disponível em setembro? Desculpem a pergunta é que essas datas confundem!

**Comment by robmachado** [comment]
Pois é confundem não só você a nós também !
A resposta a sua pergunta é .... EU TAMBÉM NÃO SEI !!  Estamos nadando em águas escuras e lamacentas tateando para achar alguma coisa.
Ou seja vamos testando para ver o que acontece, mas a data de setembro me parece a mais provável

**Comment by devsabrinaelias** [comment]
Olá, apenas passando um retorno, eu fiz hoje uma nota em produção com base na nova técnica e não houve rejeição pelo SEFAZ a nota foi emitida normalmente.

**Comment by gersonfs** [comment]
Também já usei o pagamento PIX em produção e passou. Mas ao visualizar o cupom inteiro no site do sefaz só não aparece a descrição da forma de pagamento.

**Comment by brasvidal** [comment]
> Também já usei o pagamento PIX em produção e passou. Mas ao visualizar o cupom inteiro no site do sefaz só não aparece a descrição da forma de pagamento.

Cola o XML do cupom ae :D

**Comment by gersonfs** [comment]
> > Também já usei o pagamento PIX em produção e passou. Mas ao visualizar o cupom inteiro no site do sefaz só não aparece a descrição da forma de pagamento.
> 
> Cola o XML do cupom ae :D

Não posso colar todo ele, mas sim a parte que determina o pagamento do tipo PIX. tPag 17 é o pix.
`<pag><detPag><tPag>17</tPag><vPag>10.00</vPag></detPag></pag>`

**Comment by gersonfs** [comment]
> ![image](https://user-images.githubusercontent.com/3345082/119674149-5635fe80-be12-11eb-8884-c21f27830bde.png)
> Tentou colocar a descrição no xPag?

Não, pelo que recordo a descrição seria somente no tPag=99 correto? Mas eu usei o 17. Mas nem esquenta, o site do sefaz deve estar em atualização ainda. O que interessa é que a forma de pagamento passou, no cupom eu fiz sair certo. Obrigado!

**Comment by brasvidal** [comment]
> > ![image](https://user-images.githubusercontent.com/3345082/119674149-5635fe80-be12-11eb-8884-c21f27830bde.png)
> > Tentou colocar a descrição no xPag?
> 
> Não, pelo que recordo a descrição seria somente no tPag=99 correto? Mas eu usei o 17. Mas nem esquenta, o site do sefaz deve estar em atualização ainda. O que interessa é que a forma de pagamento passou, no cupom eu fiz sair certo. Obrigado!

Ah, se vc ta usando o 17 deve ser nos teus schemas, ele que grava o nome da finalizadora na impressão. Tenta atualizar os schemas para testar.
